tests/test_common_decorators.py

```diff
@@ -126,3 +130,95 @@ def func(name, **options):
         return options
 
     assert _extract_arguments(func) == ["name", *_KNOWN_SIG_GLOBAL_PARAMETERS]
+
+
+def test_experimental_decorator_as_standalone_decorator_of_function_without_arguments():
+    test_app = typer.Typer()
+
+    @test_app.command("foo")
+    @with_experimental_behaviour()
+    def func(**options):
+        return options
+
+    assert _extract_arguments(func) == ["experimental"]
+    assert func(experimental=True) == dict(experimental=True)
+
+
+def test_experimental_decorator_as_standalone_decorator_of_function_with_arguments():
+    test_app = typer.Typer()
+
+    @test_app.command("foo")
+    @with_experimental_behaviour()
+    def func(name: str, **options):
+        return options
+
+    assert _extract_arguments(func) == ["name", "experimental"]
+    assert func(name="solaris", experimental=True) == dict(experimental=True)
+
+
+def test_experimental_decorator_below_global_options_decorator_for_function_without_arguments():
+    test_app = typer.Typer()
+
+    @test_app.command("foo")
+    @global_options
+    @with_experimental_behaviour()
```

Comment:
Position of `--experimental` flag in `--help` depends on a position of the decorator. Is it ok?

Reply:
I don't think it does matter much